src/ray/gcs/gcs_server/gcs_server_io_context_policy.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <type_traits>
+
+#include "ray/common/ray_syncer/ray_syncer.h"
+#include "ray/gcs/gcs_server/gcs_task_manager.h"
+#include "ray/gcs/pubsub/gcs_pub_sub.h"
+#include "ray/util/array.h"
+
+namespace ray {
+namespace gcs {
+
+struct GcsServerIoContextPolicy {
+  GcsServerIoContextPolicy() = delete;
+
+  // IoContext name for each handler.
+  // If a class needs a dedicated io context, it should be specialized here.
+  // If a class does NOT have a dedicated io context, returns -1;
```

Comment:
where is the code that returns -1?

Reply:
None today. later we will have classes assigned to default ioctx.